src/sentry/static/sentry/app/views/performance/transactionSummary/durationChart.tsx

```diff
@@ -48,17 +51,45 @@ const YAXIS_VALUES = ['p50()', 'p75()', 'p95()', 'p99()', 'p100()'];
  * percentiles over the past 7 days
  */
 class DurationChart extends React.Component<Props> {
+  handleLegendSelectChanged = (legendChange, _event) => {
+    const {location} = this.props;
+    const {selected} = legendChange;
+    const unselected = Object.keys(selected).filter(key => !selected[key]);
+
+    const to = {
+      ...location,
+      query: {
+        ...location.query,
+        unselectedSeries: unselected,
+      },
+    };
+    browserHistory.push(to);
+  };
+
   render() {
     const {
       api,
       project,
       environment,
+      location,
       organization,
       query,
       statsPeriod,
       router,
     } = this.props;
 
+    const {
+      query: {unselectedSeries = []},
+    } = location;
+    const seriesSelection = {};
+    if (typeof unselectedSeries === 'string') {
+      seriesSelection[unselectedSeries] = false;
+    } else {
+      for (const unselected of unselectedSeries) {
+        seriesSelection[unselected] = false;
+      }
+    }
```

Comment:
```suggestion
    const unselectedSeries = location.query.unselectedSeries || [];
    const unselectedMetrics = Array.isArray(unselectedSeries) ? unselectedSeries : [unselectedSeries];
    const seriesSelection = unselectedMetrics.reduce((acc, key) => {
      acc[key] = false;
      return acc;
    }, {});
```

🏌️ This is a bit shorter and should be functionally equivalent.

Reply:
is there a preference to use methods like map/reduce over for loops? Or is it just because its shorter?

Reply:
I bias towards map/reduce because its shorter and generally avoids in-place mutation which is problematic in react and can cause missing re-renders.